### PR TITLE
fix: add mTLS for metrics exporter

### PR DESCRIPTION
**Pull request by xuan-cao-swi**
Closes https://github.com/open-telemetry/opentelemetry-ruby/issues/1688

### CODE REVIEW

File: exporter/otlp-metrics/test/opentelemetry/exporter/otlp/metrics_exporter_test.rb [hunk 1]
```diff
@@ -97,6 +106,8 @@
     it 'prefers explicit parameters rather than the environment' do
       exp = OpenTelemetry::TestHelpers.with_env('OTEL_EXPORTER_OTLP_ENDPOINT' => 'https://localhost:1234',
                                                 'OTEL_EXPORTER_OTLP_CERTIFICATE' => '/foo/bar',
+                                                'OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE' => METRICS_CLIENT_CERT_A_PATH,
+                                                'OTEL_EXPORTER_OTLP_CLIENT_KEY' => METRICS_CLIENT_CERT_A_PATH,
```
Comment (kaylareopelle) [inline exporter/otlp-metrics/test/opentelemetry/exporter/otlp/metrics_exporter_test.rb:110]: For this test, I think we need a second certificate to verify `client_certificate_file` and `client_key_file` passed as arguments supersede the environment variables. 

https://github.com/open-telemetry/opentelemetry-ruby/blob/d9324e03f8520c35baf1fcaeb0b96d50ba4e923c/exporter/otlp/test/opentelemetry/exporter/otlp/exporter_test.rb#L110-L140

**Review by kaylareopelle — CHANGES_REQUESTED** [review]
Thanks for picking this up, Xuan! One small thing with the tests.

File: exporter/otlp-metrics/test/opentelemetry/exporter/otlp/metrics_exporter_test.rb (hunk 1 above)
Comment (xuan-cao-swi) [reply exporter/otlp-metrics/test/opentelemetry/exporter/otlp/metrics_exporter_test.rb:113]: Thanks, I have added that

**Review by kaylareopelle — APPROVED** [review]

File: exporter/otlp-metrics/README.md [hunk 2]
```diff
@@ -67,11 +67,12 @@ The collector exporter can be configured explicitly in code, or via environment
 | ------------------- | -------------------------------------------- | ----------------------------------- |
 | `endpoint:`         | `OTEL_EXPORTER_OTLP_ENDPOINT`                | `"http://localhost:4318/v1/metrics"` |
 | `certificate_file: `| `OTEL_EXPORTER_OTLP_CERTIFICATE`             |                                     |
+| `client_certificate_file: `| `OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE` |                                   |
+| `certificate_file: `| `OTEL_EXPORTER_OTLP_CLIENT_KEY`              |                                     |
```
Comment (bdewater) [inline exporter/otlp-metrics/README.md:71]:
```suggestion
| `client_key:`       | `OTEL_EXPORTER_OTLP_CLIENT_KEY`              |                                     |
```

**Review by mwear — APPROVED** [review]
LGTM. Can you fix the conflicts?

**Pull request merged**